File: cloud-collect-core/src/main/java/com/windea/demo/cloudcollect/core/validation/validator/PasswordValidator.java
package com.windea.demo.cloudcollect.core.validation.validator;

import com.windea.demo.cloudcollect.core.validation.annotation.Password;
import org.springframework.beans.factory.annotation.Value;

import javax.validation.ConstraintValidator;
import javax.validation.ConstraintValidatorContext;

/**
 * 密码的校验器。
 */
public class PasswordValidator implements ConstraintValidator<Password, String> {
	@Value("${com.windea.validation.validPassword}")
	private String validPassword;


	@Override
	public void initialize(Password constraintAnnotation) {
	}

	@Override
	public boolean isValid(String value, ConstraintValidatorContext context) {
		//6~16位的字母、数字和下划线，以字母开头。
		return value.matches(validPassword);
	}
}

File: cloud-collect-core/src/main/kotlin/com/windea/demo/cloudcollect/core/validation/validator/PasswordValidator.kt
package com.windea.demo.cloudcollect.core.validation.validator

import com.windea.demo.cloudcollect.core.properties.*
import com.windea.demo.cloudcollect.core.validation.annotation.*
import org.springframework.stereotype.*
import javax.validation.*

/**密码的校验器。*/
@Component
class PasswordValidator(
	private val validationProperties: ValidationProperties
) : ConstraintValidator<Password, String> {
	override fun isValid(value: String, context: ConstraintValidatorContext): Boolean {
		//6~16位的字母、数字和下划线，以字母开头。
		return value.matches(validationProperties.password.toRegex())
	}
}
